src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/iOS/AuthenticationAgentUIViewController.cs

```diff
@@ -152,8 +150,8 @@ public override void ViewDidLoad()
 
         private void CancelAuthentication(object sender, EventArgs e)
         {
-            callbackMethod(new AuthorizationResult(AuthorizationStatus.UserCancel, null));
             this.DismissViewController(true, null);
+            callbackMethod(new AuthorizationResult(AuthorizationStatus.UserCancel, null));
```

Comment:
isn't the override on the DismissViewController a callback to a method which will be ensure to be completed after dismissal of the viewcontroller? (see below) - I think the right thing to do is to :

this.DismissViewController(true, new AuthorizationResult( ...)

Reply:
I have a few questions about this for you...I'll swing by tomorrow. 🐈